morpheus/models/dfencoder/autoencoder.py

```diff
@@ -0,0 +1,1129 @@
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#
+#     * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided
+#        with the distribution.
+#
+#     * Neither the name of the dfencoder Developers nor the names of any
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import gc
+from collections import OrderedDict
+
+import numpy as np
+import pandas as pd
+import torch
+import tqdm
+
+from .dataframe import EncoderDataFrame
+from .logging import BasicLogger, IpynbLogger, TensorboardXLogger
+from .scalers import GaussRankScaler, NullScaler, StandardScaler, ModifiedScaler
+
+
+def ohe(input_vector, dim, device="cpu"):
+    """Does one-hot encoding of input vector."""
+    batch_size = len(input_vector)
+    nb_digits = dim
+
+    y = input_vector.reshape(-1, 1)
+    y_onehot = torch.FloatTensor(batch_size, nb_digits).to(device)
+
+    y_onehot.zero_()
+    y_onehot.scatter_(1, y, 1)
+
+    return y_onehot
+
+
+def compute_embedding_size(n_categories):
+    """
+    Applies a standard formula to choose the number of feature embeddings
+    to use in a given embedding layers.
+
+    n_categories is the number of unique categories in a column.
+    """
+    val = min(600, round(1.6 * n_categories**0.56))
+    return int(val)
+
+
+class CompleteLayer(torch.nn.Module):
+    """
+    Impliments a layer with linear transformation
+    and optional activation and dropout."""
+
+    def __init__(self, in_dim, out_dim, activation=None, dropout=None, *args, **kwargs):
+        super(CompleteLayer, self).__init__(*args, **kwargs)
+        self.layers = []
+        linear = torch.nn.Linear(in_dim, out_dim)
+        self.layers.append(linear)
+        self.add_module('linear_layer', linear)
+
+        self.activation = activation
```

Comment:
This is a change I added, fixes an obvious bug in the `interpret_activation` method when `act=None`